relay-server/src/utils/profile.rs

```diff
@@ -287,6 +287,110 @@ pub fn parse_typescript_profile(item: &mut Item) -> Result<(), ProfileError> {
     Ok(())
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+pub struct RustDebugMeta {
+    pub images: Vec<SymbolicDebugImage>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct RustProfile {
+    duration_ns: u64,
+    platform: String,
+    architecture: String,
+    // string representation of Uuid v4
+    trace_id: EventId,
+    transaction_name: String,
+    // string representation of Uuid v4
+    transaction_id: EventId,
+    // string representation of Uuid v4
+    profile_id: EventId,
+    sampled_profile: RustSampledProfile,
+    device_os_name: String,
+    device_os_version: String,
+    version_name: String,
+    version_code: String,
+    debug_meta: RustDebugMeta,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct RustSampledProfile {
+    pub start_time_nanos: u64,
+    pub start_time_secs: u64,
+    pub duration_nanos: u64,
+    pub samples: Vec<RustSample>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct RustSample {
+    pub frames: Vec<RustFrame>,
+    pub thread_name: String,
+    pub thread_id: u64,
+    pub nanos_relative_to_start: u64,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct RustFrame {
+    pub instruction_addr: String,
+}
+
+/// Represents a symbolic debug image.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct SymbolicDebugImage {
+    #[serde(rename = "type")]
+    pub image_type: String,
+    /// Path and name of the image file (required).
+    ///
+    /// The absolute path to the dynamic library or executable. This helps to locate the file if it is missing on Sentry.
+    /// This is also called `code_file`.
+    pub name: String,
+    /// The optional CPU architecture of the debug image.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub arch: Option<String>,
+    /// Starting memory address of the image (required).
+    ///
+    /// Memory address, at which the image is mounted in the virtual address space of the process.
+    pub image_addr: String,
+    /// Size of the image in bytes (required).
+    ///
+    /// The size of the image in virtual memory.
+    pub image_size: u64,
+    /// Loading address in virtual memory.
+    ///
+    /// Preferred load address of the image in virtual memory, as declared in the headers of the
+    /// image. When loading an image, the operating system may still choose to place it at a
+    /// different address.
+    ///
+    /// Symbols and addresses in the native image are always relative to the start of the image and do not consider the preferred load address. It is merely a hint to the loader.
+    pub image_vmaddr: Option<String>,
+    /// Unique debug identifier of the image.
+    ///
+    /// This is also called `debug_id`.
+    pub id: String,
+
+    /// Optional identifier of the code file.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub code_id: Option<String>,
```

Comment:
Import from the `debugid` crate.

```suggestion
    pub code_id: Option<CodeId>,
```

Reply:
marking this as resolved as I merged this type with `NativeDebugImage` shared with `ios`